gusto/diagnostics.py

```diff
@@ -1033,12 +1033,30 @@ def setup(self, domain, state_fields):
             field1 = state_fields(self.field_name1)
             field2 = state_fields(self.field_name2, space=field1.function_space(),
                                   pick_up=True, dump=False)
-            # By default set this new field to the current value
-            # This may be overwritten if picking up from a checkpoint
```

Comment:
I think we still want these deleted comments in here...
and this raises the point that we should make sure to comment our changes clearly so that we can still understand this change in 6 months time...

I suggest adding something like (you can use your own words!):
```
# The initial value for this field may not have been set yet, so we should postpone setting
# the initial field until the first time that the compute method is called
```